### PR TITLE
Implement From<Local> for Place and PlaceBase

### PR DESCRIPTION
**Pull request by spastorino**
r? @oli-obk 
More tiny bits of Place 2.0 moved into master

### CODE REVIEW

File: src/librustc/mir/mod.rs [hunk 1]
```diff
@@ -2096,6 +2096,18 @@ impl<'tcx> Place<'tcx> {
     }
 }
 
+impl From<Local> for Place<'tcx> {
```
Comment (Centril) [inline src/librustc/mir/mod.rs]:
```suggestion
impl From<Local> for Place<'_> {
```

File: src/librustc/mir/mod.rs [hunk 2]
```diff
@@ -2096,6 +2096,18 @@ impl<'tcx> Place<'tcx> {
...
+    fn from(local: Local) -> Place<'tcx> {
```
Comment (Centril) [inline src/librustc/mir/mod.rs]:
```suggestion
    fn from(local: Local) -> Self {
```

File: src/librustc/mir/mod.rs [hunk 3]
```diff
@@ -2096,6 +2096,18 @@ impl<'tcx> Place<'tcx> {
...
+        Place::Base(PlaceBase::from(local))
```
Comment (Centril) [inline src/librustc/mir/mod.rs]:
```suggestion
        Place::Base(local.into())
```

File: src/librustc/mir/mod.rs [hunk 4]
```diff
@@ -2096,6 +2096,18 @@ impl<'tcx> Place<'tcx> {
...
+    }
+}
+
+impl From<Local> for PlaceBase<'tcx> {
```
Comment (Centril) [inline src/librustc/mir/mod.rs]: Same idea here.

**Review by oli-obk — COMMENTED** [review]
@bors r+

**Pull request merged**